corehq/apps/data_dictionary/views.py

```diff
@@ -176,8 +176,11 @@ def deprecate_or_restore_case_type(request, domain, case_type_name):
 @requires_privilege_with_fallback(privileges.DATA_DICTIONARY)
 @require_permission(HqPermissions.edit_data_dict)
 def delete_case_type(request, domain, case_type_name):
-    case_type_obj = CaseType.objects.get(domain=domain, name=case_type_name)
-    case_type_obj.delete()
+    try:
+        case_type_obj = CaseType.objects.get(domain=domain, name=case_type_name)
+        case_type_obj.delete()
+    except CaseType.DoesNotExist:
+        return JsonResponse({'status': 'failed', 'message': 'Case type does not exist'})
```

Comment:
nit: translation?

Reply:
This message won't be shown on the front-end, as there is already a generic error message that gets shown. This is more a debugging response, however we could either:
1. Completely remove the message param
2. Translate the message and show it on the front-end

@mkangia I'm leaning more towards the first option, but I'm curious to hear your thoughts?

Reply:
> Completely remove the message param

If its not being used then definitely remove it.

What message does the user see though?

> Translate the message and show it on the front-end

We should go with whatever is consistent with the rest of the UI.

Reply:
> If its not being used then definitely remove it.

Will go ahead and remove the param.

> What message does the user see though?

The user will see [this](https://github.com/dimagi/commcare-hq/blob/27cce7c6581b40ec1447e5e49b4320a990485236/corehq/apps/data_dictionary/templates/data_dictionary/base.html#L137) message.

Reply:
Cool thank you